exporter/exporter.go

```diff
@@ -0,0 +1,180 @@
+package exporter
+
+import (
+	"io"
+	"encoding/json"
+	"strconv"
+
+	"github.com/google/cayley/graph"
+)
+
+type Exporter struct {
+	wr io.Writer
+	qstore graph.QuadStore
+	err error
+	count int32
+}
+
+func NewExporter(writer io.Writer, qstore graph.QuadStore) *Exporter {
```

Comment:
One thought would be to have a `NewExporterForIterator` -- given a quad-`graph.Iterator` -- so that applications might dump subgraphs. This functions would fill this in with `qstore.QuadsAllIterator()` as below.

Quad iterators exist (you found one :) ) but are hard to generate in the query languages -- for now, at least. But they could still be done by hand/in the future.

Taking quad iterators is perfectly sensible for the exporter package, IMO